src/components/Explore/SearchBar.tsx

```diff
@@ -1,54 +1,85 @@
-import useTheme from 'hooks/useTheme'
-import { useState } from 'react'
-import { Search } from 'react-feather'
+import { useRef, useState } from 'react'
 import styled from 'styled-components/macro'
 
+import searchIcon from './search.svg'
+import xIcon from './x.svg'
+
 export const SMALL_MEDIA_BREAKPOINT = '580px'
 
-const StyledSearchBar = styled.div<{ focused: boolean }>`
+const SearchBarWrapper = styled.div`
+  position: relative;
   display: flex;
   flex: 1;
-  align-items: center;
-  gap: 8px;
-  border-radius: 12px;
-  background-color: ${({ theme, focused }) => (focused ? theme.bg2 : theme.bg0)};
-  color: ${({ theme }) => theme.text2};
-  font-size: 16px;
-  padding: 0px 12px;
 `
-const SearchInput = styled.input`
-  background-color: transparent;
+
+const SearchInput = styled.input<{ expanded: boolean }>`
+  background: no-repeat scroll 7px 7px;
+  background-image: url(${searchIcon});
+  background-size: 20px 20px;
+  background-position: 12px center;
+  background-color: ${({ theme }) => theme.bg0};
+  left: 10px;
+  border-radius: 12px;
+  align-items: center;
   border: none;
-  width: 90%;
+  height: 100%;
+  width: ${({ expanded }) => (expanded ? '100%' : '44px')};
   font-size: 16px;
+  padding-left: 40px;
   color: ${({ theme }) => theme.text2};
+  animation: ${({ expanded }) => expanded && 'expand 0.8s'};
+  @keyframes expand {
+    from {
+      width: 0%;
+    }
+    to {
+      width: 100%;
+    }
+  }
 
   :focus {
     outline: none;
     background-color: ${({ theme }) => theme.bg2};
   }
   ::placeholder {
-    color: ${({ theme }) => theme.text3};
+    color: ${({ theme, expanded }) => (expanded ? theme.text3 : 'transparent')};
     @media only screen and (max-width: ${SMALL_MEDIA_BREAKPOINT}) {
       color: transparent;
     }
   }
+  ::-webkit-search-cancel-button {
+    -webkit-appearance: none;
+    appearance: none;
+    height: 24px;
```

Comment:
you need the height, width and background size? i think make a var for the 24 tho

Reply:
for some reason yes, when i remove either/or, the framing gets weird haha